sycl/include/sycl/property_list.hpp

```diff
@@ -46,9 +45,7 @@ class property_list : protected detail::PropertyListBase {
 
   template <typename PropT> PropT get_property() const {
     if (!has_property<PropT>())
-      throw sycl::invalid_object_error("The property is not found",
-                                       PI_ERROR_INVALID_VALUE);
-
+      throw std::invalid_argument("The property is not found");
```

Comment:
Yes, but that requires "#include <sycl/exception.hpp>" which will cause a cyclic dependency issue.
Note that in the exception.hpp, we have to #include <sycl/context.hpp> which in turn include this file.
In this dependency cycle, each object has to be instantiated so we can't solve this circular dependency with forward declarations. We have to break it by throwing a std::invalid_argument instead of sycl::exception.

Reply:
There are ways around it. In this case, you could have some function, let's call it `exception_helper`, that throws from within the library. It's not perfect, but we need to follow the specification.

Reply:
@steffenlarsen 
I created exception_helper.hpp /cpp to isolate throwing.